hathor/consensus.py

```diff
@@ -242,23 +243,12 @@ def update_voided_info(self, block: Block) -> None:
                     # we need to check that block is not voided.
                     meta = block.get_metadata()
                     if not meta.voided_by:
-                        storage._best_block_tips = [block.hash]
                         self.log.debug('index new winner block', height=meta.height, block=block.hash_hex)
                         # We update the height cache index with the new winner chain
                         storage.update_block_height_cache_new_chain(meta.height, block)
+                        storage.update_best_block_tips_cache([block.hash])
                 else:
-                    storage._best_block_tips = [blk.hash for blk in heads]
-                    # XXX Is it safe to select one of the heads?
-                    best_block = heads[0]
-                    assert best_block.hash is not None
-                    best_meta = best_block.get_metadata()
-                    self.log.debug('index previous best block', height=best_meta.height, block=best_block.hash_hex)
-                    storage.add_new_to_block_height_index(best_meta.height, best_block.hash)
-
-        # Uncomment the following lines to check that the cache update is working properly.
-        # You shouldn't run this test in production because it dampens performance.
-        #     v = storage.get_best_block_tips(skip_cache=True)
-        #     assert v == storage._best_block_tips
+                    storage.update_best_block_tips_cache([not_none(blk.hash) for blk in heads])
```

Comment:
No, it isn't necessary. I'm not entirely sure if we have to update the height cache, but since those blocks are voided, when one of them is eventually unvoided the correct block will be added to the height cache.

Reply:
We finished the discussion and you're right we don't need this piece of code.